src/main/java/org/opensearch/neuralsearch/query/NeuralKNNQuery.java

```diff
@@ -21,9 +21,11 @@
 @Getter
 public class NeuralKNNQuery extends Query {
     private final Query knnQuery;
+    private final String originalQueryText;
 
-    public NeuralKNNQuery(Query knnQuery) {
+    public NeuralKNNQuery(Query knnQuery, String originalQueryText) {
         this.knnQuery = knnQuery;
+        this.originalQueryText = originalQueryText;
```

Comment:
Shall we add BWC test to validate that it does not break anything in bwc environment?

Reply:
Sure, will have a check on bwc side